### PR TITLE
ability to pass a component in light prop

### PR DESCRIPTION
**Pull request by idiglove**
the light prop could be like this:
`light={sun}`

The styling of the image component should come from where it was called

The play icon will be positioned absolute if `light` is a component 

image

refers to #1320

### CODE REVIEW

File: src/Preview.js [hunk 1]
```diff
@@ -31,6 +32,10 @@ export default class Preview extends Component {
       this.setState({ image: light })
       return
     }
+    if (typeof light !== "string" && typeof light !== "boolean") {
+      this.setState({ previewImage: { isComponent: true, component: light } });
+      return;
+    }
```
Comment (n3llym) [inline src/Preview.js:38]: I'm not sure it's a good idea to assume than if it isn't a string or boolean then it is definitely a component.  Maybe you could try 
```
 else if (React.isValidElement(light) {
        this.setState({ previewImage: { isComponent: true, component: light } });
	return;
}
```

File: src/Preview.js [hunk 2]
```diff
@@ -99,6 +106,7 @@ export default class Preview extends Component {
         onKeyPress={this.handleKeyPress}
       >
         {playIcon || defaultPlayIcon}
+        {previewImage.isComponent && previewImage.component}
       </div>
```
Comment (n3llym) [inline src/Preview.js:110]: If you switch the order of these you don't need to add the z-index to the shadow style.
i.e 
```
{previewImage.isComponent && previewImage.component}
{playIcon || defaultPlayIcon}
```

Comment (n3llym) [inline src/Preview.js:110]: If you switch the order of these you don't need to add the z-index to the shadow style.
i.e 
```
{previewImage.isComponent && previewImage.component}
{playIcon || defaultPlayIcon}
```

**Review by n3llym — COMMENTED** [review]
Thanks @idiglove, and good job. It would be great to get this added!

Just a few things I noticed I've added some inline comments as well as the below.

1. You need to add the new type in base.d.ts i.e `light?: boolean | string | ReactElement`
2. You need to update the light prop type in props.js i.e` light: oneOfType([bool, string, object])`
3. Also don’t forget to run lint and run the tests!

**Comment by idiglove** [comment]
Thanks @n3llym I've done the changes, linted and tested. Appreciate your review!

**Comment by cookpete** [comment]
I tidied things up a bit to remove the need for the extra stuff in `state`. Nice work!

**Comment by cookpete** [comment]
Published in `2.11.0`. Sorry for the delay.

**Pull request merged**